forest/bonsai/tests/test_simulate_gps_data.py

```diff
@@ -0,0 +1,27 @@
+"""Tests for simulate_gps_data module"""
+
+import pytest
+
+from forest.bonsai.simulate_gps_data import get_path
+
+coords1 = (51.458498, -2.59638)
+coords2 = (51.457619, -2.608466)
+coords3 = (51.458492, -2.59635)
+api_key = "5b3ce3597851110001cf6248551c505f7c61488a887356ff5ea924d5"
```

Comment:
There should be no actual API keys or other secrets hardcoded in the source code. I recommend disabling this key in your Openrouteservice account since it has been exposed.

Also, unit tests should have no dependencies on external services. You can handle this by using mock objects (some examples here: https://nedbatchelder.com/text/test3.html)

Reply:
I am not sure how to do that. Without using the api key or the external service of openrouteservice how can I call the function inside the tests?

Reply:
You have to use a test double or mock the openrouteservice client so that `directions()` call would return the data that would have been returned from the web API. This is described in the presentation I've linked above (just search for "mock") and there's more info here: https://github.com/pytest-dev/pytest-mock

Reply:
Ok thanks, I think it should be fine now.